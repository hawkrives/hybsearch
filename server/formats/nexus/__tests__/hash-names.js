/* eslint-env jest */

const { hashNexusTreeNames } = require('../index')

test('emydura.nexus', () => {
	let content = `#NEXUS

Begin taxa;
	Dimensions ntax=10;
		Taxlabels
			Emydura_macquarii
			Emydura_subglobosa
			Emydura_tanybaraga
			Emydura_victoriae
			;
End;
Begin trees;
	Translate
		   1 Emydura_macquarii,
		   2 Emydura_subglobosa,
		   3 Emydura_tanybaraga,
		   4 Emydura_victoriae
;
tree STATE_0 = ((1[&dmv=0.022160000660789422]:0.02229904566199236,4[&dmv=0.022160000660789422]:0.02229904566199236)[&dmv=0.022160000660789422]:0.029438265490644572,(2[&dmv=0.022160000660789422]:0.0071863359974703046,3[&dmv=0.022160000660789422]:0.0071863359974703046)[&dmv=0.022160000660789422]:0.04455097515516663)[&dmv=0.022160000660789422];
End;`

	let nameMap = {
		a: 'Emydura_macquarii__KC755183',
		b: 'Emydura_subglobosa__KC692462',
		c: 'Emydura_tanybaraga__KC755186',
		d: 'Emydura_victoriae__KC755188',
	}

	const tree = hashNexusTreeNames(content, nameMap)
	expect(tree).toMatchSnapshot()
})
