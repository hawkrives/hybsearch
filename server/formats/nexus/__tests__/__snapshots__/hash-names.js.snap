// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emydura.nexus 1`] = `
"#NEXUS

Begin taxa;
	Dimensions ntax=10;
		Taxlabels
			a
			b
			c
			d
			;
End;
Begin trees;
	Translate
		   1 a,
		   2 b,
		   3 c,
		   4 d
;
tree STATE_0 = ((1[&dmv=0.022160000660789422]:0.02229904566199236,4[&dmv=0.022160000660789422]:0.02229904566199236)[&dmv=0.022160000660789422]:0.029438265490644572,(2[&dmv=0.022160000660789422]:0.0071863359974703046,3[&dmv=0.022160000660789422]:0.0071863359974703046)[&dmv=0.022160000660789422]:0.04455097515516663)[&dmv=0.022160000660789422];
End;"
`;
